web/lib/supabase/groups.ts

```diff
@@ -124,3 +203,37 @@ export async function getPublicGroups() {
 
   return groupThings.data.map((d: { data: any }) => d.data as Group)
 }
+
+export async function getGroupBySlug(groupSlug: string) {
+  const { data } = await run(
+    db.from('groups').select('data').eq('slug', groupSlug).limit(1)
+  )
+  return data ? (data[0]?.data as Group) : null
+}
+
+export async function getGroup(groupId: string) {
+  const { data } = await run(
+    db.from('groups').select('data').eq('id', groupId).limit(1)
+  )
+  return data ? (data[0]?.data as Group) : null
+}
+
+export async function getGroupContractIds(groupId: string) {
+  const { data } = await run(
+    db.from('group_contracts').select('contract_id').eq('group_id', groupId)
+  )
+  if (data && data.length > 0) {
+    return data.map((group) => group.contract_id as string)
+  }
+  return []
+}
+
+export async function listGroupsBySlug(groupSlugs: string[]) {
+  const { data } = await run(
+    db.from('groups').select('data').in('slug', groupSlugs)
+  )
+  if (data && data.length > 0) {
+    return data.map((group) => group.data as Group)
+  }
+  return []
+}
```

Comment:
I think @IanPhilips just added a few indexes for groups and related tables. Would be good to still check that these queries are covered! (Check seed.sql or see if your db tools shows what indexes were created, or run the query in your console with explain analyze and check it is < 40ms.)